### PR TITLE
Add callback which will be called after all layers are loaded.

### PR DESCRIPTION
**Pull request by nilscb**

### CODE REVIEW

**Comment by nilscb** [comment]
This pull request introduces a callBack property which will be triggered when data in all layers are loaded.

**Comment by hkfb** [comment]
Suggest rebasing on master.

File: typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx [hunk 1]
```diff
@@ -644,3 +681,55 @@ ViewMatrixMargin.args = {
...
+                <SubsurfaceViewer {...props} />
+            </div>
+            <button onClick={handleChange}> Add layer </button>
+        </Root>
+    );
+};
+
+IsLoadedCallback.args = {
+    id: "DeckGL-Map",
+    isLoadedCallback: (isLoaded) => console.log("isLoadedCallback", isLoaded),
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx:713]: Should be more visible, like eg. a label in the map

File: typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx [hunk 2]
```diff
@@ -644,3 +681,55 @@ ViewMatrixMargin.args = {
         ],
     },
 };
+
+export const IsLoadedCallback = (args) => {
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx:685]: Please move to `SubsurfaceViewer.stories.tsx`. Let's try to deprecate the JSX stories.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx [hunk 3]
```diff
@@ -326,6 +324,34 @@ const meshMapLayerPng = {
     colorMapName: "Physics",
 };
 
+const meshMapLayerBig = {
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx:327]: Where is this used?

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx [hunk 4]
```diff
@@ -16,12 +16,32 @@ import { ExtendedLayerProps, LayerPickInfo } from "./layers/utils/layerTools";
...
 import { FeatureCollection } from "geojson";
 import { ViewFooter } from "./components/ViewFooter";
+import { styled } from "@mui/material/styles";
 
 export default {
     component: SubsurfaceViewer,
     title: "SubsurfaceViewer",
 } as ComponentMeta<typeof SubsurfaceViewer>;
 
+const PREFIX = "Default";
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:26]: This looks like boilerplate `codemod` autogenerated code copied from elsewhere. I think there's room for improvement. At least I don't think the `PREFIX` is needed.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx (hunk 3 above)
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx:327]: It is used int the story?! The point is to have a big map which takes some tine to load

Comment (hkfb) [reply typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx:327]: It's not used in any of those stories. You moved it to the .tsx file.

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx (hunk 4 above)
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx]: you're right my mistake :-)

File: typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx (hunk 3 above)
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/storybook/SubsurfaceViewer.stories.jsx]: Your're right my mistake :-)

**Review by hkfb — CHANGES_REQUESTED** [review]
I get some typescript errors:
```
packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:485:34 - error TS7006: Parameter 'args' implicitly has an 'any' type.

485 export const IsLoadedCallback = (args) => {                 
                                     ~~~~                                                 
                                                                                          
packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:491:24 - error TS2740: Type 'MapLayer' is missing the following properties from type 'WellsLayer': setSelection, setMultiSelection, getLegendData, setLegend, and 4 more.
                                                                                          
491             setLayers([meshMapLayerBig, defaultWellsLayer]);                                                                                                                     
                           ~~~~~~~~~~~~~~~

packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:499:28 - error TS7006: Parameter 'isLoaded' implicitly has an 'any' type.
                                                                                          
499         isLoadedCallback: (isLoaded) => {                                                                                                                                        
                               ~~~~~~~~                                                   
```

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx [hunk 5]
```diff
@@ -445,3 +481,61 @@ ViewStateSynchronization.argTypes = {
...
+        },
+        layers,
+    };
+
+    return (
+        <Root>
+            <div className={classes.main}>
+                <SubsurfaceViewer {...props} />
+            </div>
+            <button onClick={handleChange}> Add layer </button>
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:512]: A bit confusing that this button both adds and removes a layer when it always says "Add layer".

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx [hunk 6]
```diff
@@ -445,3 +481,64 @@ ViewStateSynchronization.argTypes = {
...
+        },
+        layers,
+    };
+
+    return (
+        <Root>
+            <div className={classes.main}>
+                <SubsurfaceViewer {...props} />
+            </div>
+            <button onClick={handleChange}> Add/remove layer </button>
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx:515]: Maybe you could use a switch or toggle instead, saying "Show map layer"?

**Review by hkfb — CHANGES_REQUESTED** [review]

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx (hunk 6 above)
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx]: I am not quite sure how to make a checkbox but I changed the text dynamically to say either "Add layer" or "Remove layer" depending.

**Review by hkfb — APPROVED** [review]

File: typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx (hunk 6 above)
Comment (nilscb) [reply typescript/packages/subsurface-viewer/src/SubsurfaceViewer.stories.tsx]: Used mui switch instead

**Pull request merged**